src/main/kotlin/no/nav/syfo/identhendelse/IdenthendelseService.kt

```diff
@@ -44,10 +45,12 @@ class IdenthendelseService(
     }
 
     // Erfaringer fra andre team tilsier at vi burde dobbeltsjekke at ting har blitt oppdatert i PDL før vi gjør endringer
-    private suspend fun checkThatPdlIsUpdated(nyIdent: PersonIdent) {
-        val pdlIdenter = pdlClient.hentIdenter(nyIdent.value)
-        if (nyIdent.value != pdlIdenter?.aktivIdent || pdlIdenter.identer.any { it.ident == nyIdent.value && it.historisk }) {
-            throw IllegalStateException("Ny ident er ikke aktiv ident i PDL")
+    private fun checkThatPdlIsUpdated(nyIdent: PersonIdent) {
+        runBlocking {
+            val pdlIdenter = pdlClient.hentIdenter(nyIdent.value)
+            if (nyIdent.value != pdlIdenter?.aktivIdent && pdlIdenter?.aktivIdentIsHistorisk(nyIdent) != true) {
```

Comment:
Skal det ikke være || i stedet for &&?

Og kanskje også != false?

Reply:
Jeg endret litt på dette sist. Det er to caser der ny ident ikke er aktiv ident:
1. Person får ny ident og den er ikke oppdatert i pdl enda - her vil vi kaste en feil:
  - `nyIdent` != `aktivIdent` && `nyIdent` er ikke ikke i lista over historiske identer (altså `!= true`). Vi kaster feil
2. Vi leser gjennom topicen fra starten og får inn en oppdatering på ident. Identene som kommer inn er derimot ikke den nyeste, fordi det vil komme en oppdatering på identen på et senere offset i topicen. Den identen vi da får inn vil aldri bli `aktivIdent` fra PDL.
  - `nyIdent` != `aktivIdent` && `nyIdent` er i lista over historiske identer `== true`, så vi kaster ikke en feil, men ting går gjennom og den oppdateres.
  
Jeg har tenkt mye på dette og _trooor_ det blir riktig logikk, men hvis dere kommer på noen andre caser så si gjerne i fra!

Reply:
Fint, da skjønner jeg. Tror vi er enige om at null-checken på pdlIdenter gjøres separat, og at aktivIdentIsHistorisk får et bedre navn 🏆